### PR TITLE
Some fixes to exec directory management

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #11842

### CODE REVIEW

**Comment by mbiebl** [comment]
"symlink dance when runtime dire" -> typo in commit message

**Comment by poettering** [comment]
> "symlink dance when runtime dire" -> typo in commit message

Fixed.

File: src/core/execute.c [hunk 1]
```diff
@@ -2157,37 +2157,43 @@ static int setup_exec_directory(
...
-                /* Don't change the owner of the configuration directory, as in the common case it is not written to by
-                 * a service, and shall not be writable. */
-                if (type == EXEC_DIRECTORY_CONFIGURATION)
-                        continue;
+                /* Lock down the access mode (we use chmod_and_chown() to make this idempotent. We don't
+                 * specifiy uid/gid here, so that path_chown_recursive() can optimize things depending on the
+                 * current setting. */
+                if (chmod_and_chown(pp ?: p, context->directories[type].mode, UID_INVALID, GID_INVALID) < 0) {
+                        r = -errno;
+                        goto fail;
```
Comment (keszybz) [inline src/core/execute.c:2195]:
```C
r = chmod_and_chown();
if (r < 0)
   goto fail;
```

**Comment by poettering** [comment]
Force pushed new version. only change is the issue @keszybz found. ptal!

**Comment by keszybz** [comment]
Centos CI TEST-24-IMPORT fails with:
Failed to transfer image: Failed to activate service 'org.freedesktop.import1': timed out (service_start_timeout=25000ms)
I suppose that's unrelated to this PR.

**Pull request merged**